src/custom/components/Header/URLWarning/index.tsx

```diff
@@ -49,26 +50,48 @@ const Wrapper = styled.div`
 
 // Announcement content: Modify this file to edit the announcement
 //  https://github.com/gnosis/cowswap/blob/configuration/config/announcements/announcements.md
-const ANNOUNCEMENTS_MARKDOWN_BASE_URL = RAW_CODE_LINK + '/configuration/config/announcements'
+const ANNOUNCEMENTS_MARKDOWN_BASE_URL = RAW_CODE_LINK + '/configuration/config'
 
-function getAnnouncementUrl(chainId: number) {
-  return `${ANNOUNCEMENTS_MARKDOWN_BASE_URL}/announcements-${chainId}.md`
+function getAnnouncementUrl(chainId: number, env?: 'production' | 'barn') {
+  return `${ANNOUNCEMENTS_MARKDOWN_BASE_URL}${env ? `/${env}` : ''}/announcements/announcements-${chainId}.md`
 }
 
-export default function URLWarning() {
-  const { chainId = ChainId.MAINNET } = useWalletInfo()
+const PRODUCTION_ENVS: (typeof environmentName)[] = ['production', 'staging', 'ens']
 
-  // Ger announcement if there's one
+function useGetAnnouncement(chainId: number): string | undefined {
+  const env = PRODUCTION_ENVS.includes(environmentName) ? 'production' : 'barn'
+
+  // Fetches global announcement
   const { file, error } = useFetchFile(getAnnouncementUrl(chainId))
+  // Fetches env announcement
+  const { file: envFile, error: envError } = useFetchFile(getAnnouncementUrl(chainId, env))
+
   const announcementText = error ? undefined : file?.trim()
-  const contentHash = announcementText ? hashCode(announcementText).toString() : undefined
 
   if (error) {
     console.error('[URLWarning] Error getting the announcement text: ', error)
   } else {
-    console.debug('[URLWarning] Announcement text', announcementText, contentHash)
+    console.debug('[URLWarning] Announcement text', announcementText)
+  }
+
+  const envAnnouncementText = envError ? undefined : envFile?.trim()
+
+  if (envError) {
+    console.error(`[URLWarning] Error getting the env ${env} announcement text: `, envError)
+  } else {
+    console.debug(`[URLWarning] Env ${env} announcement text`, envAnnouncementText)
   }
 
+  return announcementText || envAnnouncementText
```

Comment:
shouldn't be the other way around, env taking precedence?

Reply:
For me makes more sense to be able to have a GENERAL announcent, and if you want to be more specific, and ENV one. If we have both for whatever reason, more specific should always win, right?